prereise/gather/data/remap_ba_area/remap_ba_area.py

```diff
@@ -0,0 +1,270 @@
+        "CITY OF LODI - (CA)": "CISO-PGAE",
+        "CITY OF LOMPOC - (CA)": "CISO-PGAE",
+        "CITY OF PALO ALTO - (CA)": "CISO-PGAE",
+        "CITY OF SANTA CLARA - (CA)": "CISO-PGAE",
+        "CITY OF UKIAH - (CA)": "CISO-PGAE",
+        "LASSEN MUNICIPAL UTILITY DISTRICT": "CISO-PGAE",
+        "PLUMAS-SIERRA RURAL ELEC COOP": "CISO-PGAE",
+        # SCE proper & friends
+        "SOUTHERN CALIFORNIA EDISON CO": "CISO-SCE",
+        "BEAR VALLEY ELECTRIC SERVICE": "CISO-SCE",
+        "CITY OF ANAHEIM - (CA)": "CISO-SCE",
+        "CITY OF AZUSA": "CISO-SCE",
+        "CITY OF BANNING - (CA)": "CISO-SCE",
+        "CITY OF COLTON - (CA)": "CISO-SCE",
+        "CITY OF MORENO VALLEY - (CA)": "CISO-SCE",
+        "CITY OF PASADENA - (CA)": "CISO-SCE",
+        "CITY OF RIVERSIDE - (CA)": "CISO-SCE",
+        "CITY OF VERNON": "CISO-SCE",
+        "CITY OF VICTORVILLE - (CA)": "CISO-SCE",
+    }
+
+    utility_mapping = {**balancing_authorities, **caiso_subregion_names}
+    caiso_subregions = hifld_utilities.loc[
+        hifld_utilities["NAME"].isin(utility_mapping)
+    ].copy()
+    caiso_subregions["NAME"] = caiso_subregions["NAME"].map(utility_mapping)
+    caiso_subregions.rename({"NAME": "BA"}, axis="columns", inplace=True)
+    # Tag the existing geometries with their true CRS, then re-project
+    caiso_subregions = caiso_subregions.set_crs(3857, allow_override=True)
+    caiso_subregions = caiso_subregions.to_crs(4269)
+    california_joined = pd.concat([california.reset_index(), caiso_subregions])
+    # We simultaneously concatenate and dissolve
+    full_ba_map = (
+        pd.concat([ba_dissolved.reset_index(), california_joined])
+        .dissolve(by="BA")["geometry"]
+        .to_frame()
+    )
+    # Subtract the CAISO Valley Electric Association geometry from the Nevada Power geometry
+    full_ba_map.loc["NEVP", "geometry"] = full_ba_map.loc[
+        "NEVP", "geometry"
+    ].difference(full_ba_map.loc["CISO-VEA", "geometry"])
+
+    nyiso = get_nyiso_area(crs=full_ba_map.crs)
+    full_ba_map = pd.concat([full_ba_map, nyiso])
+    # add ISONE
+    full_ba_map = full_ba_map.drop("ISNE")
+    isone_df = get_isone()
+    isone_df = isone_df.assign(BA=isone_df["BA"].apply(lambda t: "ISONE-" + t))
+    full_ba_map = pd.concat([full_ba_map, isone_df.set_index("BA")])
+    # assign the four utility territories into existing CA/subregion
+    # rename TVA_LGEE to TVA, to match the source name
+    extra_map = {
+        "NV Energy": "CISO-PGAE",
+        "SCL_TPWR": "BPAT",
+        "TID": "CISO-PGAE",
+        "WAPA": "CISO-SCE",
+        "TVA_LGEE": "TVA",
+    }
+    full_ba_map = full_ba_map.reset_index().replace({"BA": extra_map}).set_index("BA")
+    full_ba_map = full_ba_map.dissolve(by="BA")
+
+    rows = full_ba_map.shape[0]
+    name_map = {**miso_name_map, **spp_name_map}
+    full_ba_map.reset_index(inplace=True)
+    full_ba_map["BA"] = full_ba_map["BA"].map(lambda x: name_map.get(x, x))
```

Comment:
Is this equivalent to `full_ba_map["BA"].replace(name_map, inplace=True)`?

Reply:
Right, it's a simpler way. Will update.